controllers/agentcluster_controller.go

```diff
@@ -132,18 +141,75 @@ func (r *AgentClusterReconciler) updateAgentClusterInstall(ctx context.Context,
 	return ctrl.Result{}, nil
 }
 
-func (r *AgentClusterReconciler) createClusterDeployment(ctx context.Context, log logrus.FieldLogger, agentCluster *capiproviderv1alpha1.AgentCluster) (ctrl.Result, error) {
-	log.Info("Creating clusterDeployment")
+func (r *AgentClusterReconciler) getUnstructuredControlPlane(ctx context.Context, log logrus.FieldLogger,
+	agentCluster *capiproviderv1alpha1.AgentCluster) (*unstructured.Unstructured, error) {
```

Comment:
Consider renaming this method to getControlPlane and return a struct with the field we require.
You can use the unstractured instance in this function but return a contgrolPlane instance with known fields

Reply:
you mean that i need to create new struct?

Reply:
i need object itself too, for the watcher. If we want to drop it than no problem, i can add unstructured to the struct too. 
As you wish but currently i need unstructured  object itself